app/controller.go

```diff
@@ -133,3 +140,86 @@ func startController(c *cli.Context) error {
 	})
 	return http.ListenAndServe(controlListener, router)
 }
+
+func checkPrerequisits(c *controller.Controller, replicas []types.Replica) error {
+	if c.IsSnapDeletionInProgress {
+		return fmt.Errorf("Snapshot deletion process is in progress, %s is getting deleted", c.SnapshotName)
+	}
+
+	rf := c.ReplicationFactor
+	if len(replicas) != rf {
+		return fmt.Errorf("Can not remove a snapshot because, RF: %v, replica count: %v", rf, len(replicas))
+	}
+
+	for _, rep := range replicas {
+		r := rep // pin it
+		if err := c.IsReplicaRW(&r); err != nil {
+			return fmt.Errorf("Can't delete snapshot, err: %v", err)
+		}
+	}
+	return nil
+}
+
+func listSnapshot(c *controller.Controller) ([]string, error) {
+	replicas := c.ListReplicas()
+	reps := []rest.Replica{}
+	for _, rep := range replicas {
+		r := rest.Replica{
+			Address: rep.Address,
+			Mode:    string(rep.Mode),
+		}
+		reps = append(reps, r)
+	}
+	snapshots, err := filterSnapshot(reps)
```

Comment:
this, as well, is time consuming call.. we, better, avoid locks when doing this call